### PR TITLE
- getBoundingClientRect error

### LINKED ISSUE #208: getBoundingClientRect issue when using TextOverflowWrapper
## Describe the bug
getBoundingClientRect of null issue when using TextOverflowWrapper
PB > Settings > Servers > System Info > Database displays console errors

## To Reproduce
1.  Login as host
2. Go to PB > Settings > Servers > System Info 
3. Open browser debugger 
4. Click on Database tab and then on Web and then again on Database tab
5. See error in debugger

## Expected behavior
This getBoundingClientRect  should not appear

## Screenshots
image

## Error log
ncaught TypeError: Cannot read property 'getBoundingClientRect' of null
    at export-bundle.js?cdv=52:1

### PR DESCRIPTION
**Pull request by Mhtshum**
Fixes #208  

## Summary
Remove 
- Wrong react lifecycle hook componentDidMount with shouldComponentUpdate
- Settimeout

### CODE REVIEW

File: src/TextOverflowWrapper/index.jsx [hunk 1]
```diff
@@ -18,9 +18,8 @@ class TextOverflowWrapper extends Component {
         return Object.assign({ maxWidth: props.maxWidth }, props.style);
     }
 
-    componentDidMount() {
+    shouldComponentUpdate() {
         //Set time out to ensure calculation happens after render
```
Comment (daguiler) [inline src/TextOverflowWrapper/index.jsx]: I think comment needs to go away too

**Comment by Mhtshum** [comment]
@daguiler , @mtrutledge, @nvisionative 
We need to revert this change as pages throughout application containing TextOverflowWrapper are crashing and  throwing this error in debugger due to tooltip:

react-dom.development.js?61bb:55 Uncaught Error: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
    at invariant (react-dom.development.js?61bb:55)
    at scheduleWork (react-dom.development.js?61bb:18406)
    at Object.enqueueSetState (react-dom.development.js?61bb:12313)
    at ProxyComponent.Component.setState (react.development.js?72d0:372)
    at ProxyComponent.value (dnn-react-common.min.js?e802:1)
    at ProxyComponent.value (react-hot-loader.development.js?c2cb:587)
    at checkShouldComponentUpdate (react-dom.development.js?61bb:12360)
    at updateClassInstance (react-dom.development.js?61bb:12809)
    at updateClassComponent (react-dom.development.js?61bb:14262)
    at beginWork (react-dom.development.js?61bb:15082)
invariant @ react-dom.development.js?61bb:55
scheduleWork @ react-dom.development.js?61bb:18406
enqueueSetState @ react-dom.development.js?61bb:12313
Component.setState @ react.development.js?72d0:372
value @ dnn-react-common.min.js?e802:1
getPortalSettings @ react-hot-loader.development.js?c2cb:587
checkShouldComponentUpdate @ react-dom.development.js?61bb:12360
updateClassInstance @ react-dom.development.js?61bb:12809
updateClassComponent @ react-dom.development.js?61bb:14262
beginWork @ react-dom.development.js?61bb:15082
performUnitOfWork @ react-dom.development.js?61bb:17820
workLoop @ react-dom.development.js?61bb:17860
renderRoot @ react-dom.development.js?61bb:17946
performWorkOnRoot @ react-dom.development.js?61bb:18837
performWork @ react-dom.development.js?61bb:18749
performSyncWork @ react-dom.development.js?61bb:18723
requestWork @ react-dom.development.js?61bb:18592
scheduleWork @ react-dom.development.js?61bb:18401
enqueueSetState @ react-dom.development.js?61bb:12313
Component.setState @ react.development.js?72d0:372
(anonymous) @ translatePageContent.jsx?0f91:46
(anonymous) @ languages.js?891f:350
success @ sf.js?cdv=52&t=0.27001377125703585:51
i @ jquery.js?cdv=52&t=0.27001377125703585:3
fireWith @ jquery.js?cdv=52&t=0.27001377125703585:3
A @ jquery.js?cdv=52&t=0.27001377125703585:11
(anonymous) @ jquery.js?cdv=52&t=0.27001377125703585:11
load (async)
send @ jquery.js?cdv=52&t=0.27001377125703585:11
ajax @ jquery.js?cdv=52&t=0.27001377125703585:11
rawCall @ sf.js?cdv=52&t=0.27001377125703585:89
call @ sf.js?cdv=52&t=0.27001377125703585:100
getsilence @ sf.js?cdv=52&t=0.27001377125703585:120
getPageList @ languageEditorService.js?83ef:62
(anonymous) @ languages.js?891f:342
(anonymous) @ index.js?b083:8
getPageList @ translatePageContent.jsx?0f91:45
getPortalSettings @ react-hot-loader.development.js?c2cb:587
init @ translatePageContent.jsx?0f91:29
getPortalSettings @ react-hot-loader.development.js?c2cb:587
TranslatePageContent @ translatePageContent.jsx?0f91:24
TranslatePageContent @ VM33556:5
constructClassInstance @ react-dom.development.js?61bb:12484
updateClassComponent @ react-dom.development.js?61bb:14255
beginWork @ react-dom.development.js?61bb:15082
performUnitOfWork @ react-dom.development.js?61bb:17820
workLoop @ react-dom.development.js?61bb:17860
renderRoot @ react-dom.development.js?61bb:17946
performWorkOnRoot @ react-dom.development.js?61bb:18837
performWork @ react-dom.development.js?61bb:18749
performSyncWork @ react-dom.development.js?61bb:18723
interactiveUpdates$1 @ react-dom.development.js?61bb:18992
interactiveUpdates @ react-dom.development.js?61bb:2169
dispatchInteractiveEvent @ react-dom.development.js?61bb:4876
react-dom.development.js?61bb:15749 The above error occurred in the <TextOverflowWrapper> component:
    in TextOverflowWrapper (created by TranslatePageContent)

**Comment by Mhtshum** [comment]
Seems that this [PR](https://github.com/romainberger/react-portal-tooltip/pull/87) still not fix the tooltip issue. I checked with react-portal-tooltip 2.4.3 version as well.

**Comment by valadas** [comment]
@Mhtshum I discovered that too, strange it fixed it in the storybook, then when trying the latest build, the issue was back in the PB. I am both investigating this and working on an alternate solution for the tooltips, but it is not related tot he TextOverflowWrapper right ?

**Comment by valadas** [comment]
I think the issue here is that the fix only got into the release branch and not in the master branchand the rest of the build process pulls from master. When I do a manual build and assemble the pieces of the puzzle, it works fine.

**Comment by Mhtshum** [comment]
Yes this PR is just re-rendering after  getBoundingClientRect calculation by using shouldComponentUpdate hook

**Pull request merged**